pkg/controller/nodes/executor.go

```diff
@@ -449,7 +451,7 @@ func (c *nodeExecutor) handleNode(ctx context.Context, dag executors.DAGStructur
 		}
 		nodeStatus.UpdatePhase(v1alpha1.NodePhaseFailed, v1.Now(), nodeStatus.GetMessage())
 		c.metrics.FailureDuration.Observe(ctx, nodeStatus.GetStartedAt().Time, nodeStatus.GetStoppedAt().Time)
-		return executors.NodeStatusFailed(fmt.Errorf(nodeStatus.GetMessage())), nil
+		return executors.NodeStatusFailed(utils.ParseExecutionError(nodeStatus.GetMessage())), nil
```

Comment:
Not a big fan, can we just keep it a strongly typed exception all the way through?

Reply:
ohh i think I will have to implement the DeepCopy?

Reply:
done